currency/forexprovider/exchangerate.host/exchangerate.go

```diff
@@ -0,0 +1,262 @@
+package exchangeratehost
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/currency/forexprovider/base"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+)
+
+// A client for the exchangerate.host API. NOTE: The format and callback
+// parameters aren't supported as they're not needed for this implementation.
+// Furthermore, the source is set to "ECB" as default
+
+const (
+	timeLayout          = "2006-01-02"
+	exchangeRateHostURL = "https://api.exchangerate.host"
+)
+
+var (
+	// DefaultSource uses the ecb for forex rates
+	DefaultSource = "ecb"
+)
+
+// Setup sets up the ExchangeRateHost config
+func (e *ExchangeRateHost) Setup(config base.Settings) error {
+	e.Name = config.Name
+	e.Enabled = config.Enabled
+	e.RESTPollingDelay = config.RESTPollingDelay
+	e.Verbose = config.Verbose
+	e.PrimaryProvider = config.PrimaryProvider
+	e.Requester = request.New(e.Name,
+		common.NewHTTPClientWithTimeout(base.DefaultTimeOut))
+	return nil
+}
+
+// GetLatestRates returns a list of forex rates based on the supplied params
+func (e *ExchangeRateHost) GetLatestRates(baseCurrency, symbols string, amount float64, places int64, source string) (*LatestRates, error) {
+	v := url.Values{}
+	if baseCurrency != "" {
+		v.Set("base", baseCurrency)
+	}
```

Comment:
this pattern is repeated many times, does it deserve a helper?

Reply:
i think it does. but right now priority is to recompile gocryptotrader which is relying on this pkg.